internal/controller/amaltheasession_controller.go

```diff
@@ -45,21 +45,19 @@ type AmaltheaSessionReconciler struct {
 	MetricsClient metricsv1beta1.PodMetricsesGetter
 }
 
-// Definitions to manage status conditions
-const (
-	// typeAvailableAmaltheaSession represents the status of the StatefulSet reconciliation
-	typeAvailableAmaltheaSession = "Available"
-	// typeDegradedAmaltheaSession represents the status used when the custom resource is deleted and the finalizer operations are must to occur.
-	typeDegradedAmaltheaSession = "Degraded"
-)
-
 // finalizers
 const secretCleanupFinalizerName = "amalthea.dev/secrets-finalizer"
 
 //+kubebuilder:rbac:groups=amalthea.dev,resources=amaltheasessions,verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups=amalthea.dev,resources=amaltheasessions/status,verbs=get;update;patch
 //+kubebuilder:rbac:groups=amalthea.dev,resources=amaltheasessions/finalizers,verbs=update
-//+kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;delete
+//+kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;delete
+//+kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;watch
+//+kubebuilder:rbac:groups=core,resources=persistentvolumeclaims,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=core,resources=services,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=apps,resources=statefulsets,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=networking.k8s.io,resources=ingresses,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=metrics.k8s.io,resources=pods,verbs=get;list;watch
```

Comment:
We dont need delete on the child resources anywhere except on secrets. The deletions are all done by parent/child relationships. Adopted secrets are deleted by the controller and need rbac. After you do this I am pretty sure you will have to rerun the make command that regenerates the manifests.

```suggestion
//+kubebuilder:rbac:groups=amalthea.dev,resources=amaltheasessions,verbs=get;list;watch;create;update;patch;delete
//+kubebuilder:rbac:groups=amalthea.dev,resources=amaltheasessions/status,verbs=get;update;patch
//+kubebuilder:rbac:groups=amalthea.dev,resources=amaltheasessions/finalizers,verbs=update
//+kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;delete
//+kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;watch
//+kubebuilder:rbac:groups=core,resources=persistentvolumeclaims,verbs=get;list;watch;create;update;patch
//+kubebuilder:rbac:groups=core,resources=services,verbs=get;list;watch;create;update;patch
//+kubebuilder:rbac:groups=apps,resources=statefulsets,verbs=get;list;watch;create;update;patch
//+kubebuilder:rbac:groups=networking.k8s.io,resources=ingresses,verbs=get;list;watch;create;update;patch
//+kubebuilder:rbac:groups=metrics.k8s.io,resources=pods,verbs=get;list;watch
```

Reply:
Fixed